src/lib.rs

```diff
@@ -986,6 +986,103 @@
 //! let server_setup = ServerSetup::<Default, YourRemoteKey>::new_with_key(&mut OsRng, keypair);
 //! ```
 //!
+//! ## Custom KSF and Parameters
+//!
+//! An application might want to use a custom KSF (Key Stretching Function)
+//! that's not supported directly by this crate. The maintainer of the said KSF
+//! or of the application itself can implement the [`Ksf`](ksf::Ksf) trait to
+//! use it with `opaque-ke`. [`scrypt`] is used for this example, but any KSF
+//! can be used.
+//! ```
+//! # use generic_array::GenericArray;
+//! #[derive(Default)]
+//! struct CustomKsf(scrypt::Params);
+//!
+//! // The Ksf trait must be implemented to be used in the ciphersuite.
+//! impl opaque_ke::ksf::Ksf for CustomKsf {
+//!     fn hash<L: generic_array::ArrayLength<u8>>(
+//!         &self,
+//!         input: GenericArray<u8, L>,
+//!     ) -> Result<GenericArray<u8, L>, opaque_ke::errors::InternalError> {
+//!         let mut output = GenericArray::<u8, L>::default();
+//!         scrypt::scrypt(&input, &[], &self.0, &mut output)
+//!             .map_err(|_| opaque_ke::errors::InternalError::KsfError)?;
+//!
+//!         Ok(output)
+//!     }
+//! }
+//! ```
+//!
+//! It is also possible to override the default derivation parameters that are
+//! used by the KSF during registration and login. This can be especially
+//! helpful if the `Ksf` trait is already implemented.
+//! ```
+//! # use opaque_ke::CipherSuite;
+//! # use opaque_ke::ClientRegistration;
+//! # use opaque_ke::ClientRegistrationFinishParameters;
+//! # use opaque_ke::ServerSetup;
+//! # use opaque_ke::errors::ProtocolError;
+//! # use rand::rngs::OsRng;
+//! # use rand::RngCore;
+//! # use std::default::Default;
+//! #[cfg(feature = "argon2")]
```

Comment:
```suggestion
//! # #[cfg(feature = "argon2")]
```
That `cfg` can be hidden too.